cmd/helm/flags.go

```diff
@@ -112,33 +115,87 @@ func (o *outputValue) Set(s string) error {
 }
 
 func bindPostRenderFlag(cmd *cobra.Command, varRef *postrender.PostRenderer) {
-	cmd.Flags().Var(&postRenderer{varRef}, postRenderFlag, "the path to an executable to be used for post rendering. If it exists in $PATH, the binary will be used, otherwise it will try to look for the executable at the given path")
+	p := &postRendererOptions{varRef, "", []string{}}
+	cmd.Flags().Var(&postRendererString{p}, postRenderFlag, "the path to an executable to be used for post rendering. If it exists in $PATH, the binary will be used, otherwise it will try to look for the executable at the given path")
+	cmd.Flags().Var(&postRendererArgsSlice{p}, postRenderArgsFlag, "an argument to the post-renderer (can specify multiple)")
+}
+
+type postRendererOptions struct {
+	renderer   *postrender.PostRenderer
+	binaryPath string
+	args       []string
 }
 
-type postRenderer struct {
-	renderer *postrender.PostRenderer
+type postRendererString struct {
+	options *postRendererOptions
 }
 
-func (p postRenderer) String() string {
-	return "exec"
+func (p *postRendererString) String() string {
+	return p.options.binaryPath
 }
 
-func (p postRenderer) Type() string {
-	return "postrenderer"
+func (p *postRendererString) Type() string {
+	return "postRendererString"
 }
 
-func (p postRenderer) Set(s string) error {
-	if s == "" {
+func (p *postRendererString) Set(val string) error {
+	if val == "" {
 		return nil
 	}
-	pr, err := postrender.NewExec(s)
+	p.options.binaryPath = val
+	pr, err := postrender.NewExec(p.options.binaryPath, p.options.args...)
 	if err != nil {
 		return err
 	}
-	*p.renderer = pr
+	*p.options.renderer = pr
 	return nil
 }
 
+type postRendererArgsSlice struct {
+	options *postRendererOptions
+}
+
+func (p *postRendererArgsSlice) String() string {
+	return "[" + strings.Join(p.options.args, ",") + "]"
+}
+
+func (p *postRendererArgsSlice) Type() string {
+	return "postRendererArgsSlice"
+}
+
+func (p *postRendererArgsSlice) Set(val string) error {
+	if val == "" {
+		return nil
+	}
+
```

Comment:
I think that for the arguments to the post-renderer, an empty argument could be valid.
I don't have an example, but a post-renderer defined by a user may accept empty arguments specified like this:
```
$ bin/helm template test bitnami/nginx --post-renderer myrenderer --post-renderer-args ''

```
Notice that is this different than forgetting to add an argument, in which case an error will be shown:
```
$ bin/helm template test bitnami/nginx --post-renderer echo --post-renderer-args
Error: flag needs an argument: --post-renderer-args
```
So, what do you think of removing the check for `val == ""` here?

Reply:
i got !
very thanks your  detailed and comprehensive advice, that help me  learn much more code skills.
many thanks!  and `helm` is awesome golang project !